eventorganizer.api/src/main/java/com/event/organizer/api/controller/EventController.java

```diff
@@ -86,7 +87,49 @@ public Event getEventById(long eventId, Principal principal) {
         return eventService.getEventById(eventId, principal.getName());
     }
 
-    private Event getEventModel(EventRequestDto eventRequestDto) throws EventOrganizerException {
+    @GetMapping("/getThisWeeksEvents")
+    public List<Event> getThisWeeksEvents() {
+        return eventService.getThisWeeksEvents();
+    }
+
+    @GetMapping("/getThisMonthsEvents")
+    public List<Event> getThisMonthsEvents() {
+        return eventService.getThisMonthsEvents();
+    }
+
+    @GetMapping("getEventsByUserLocation")
+    public List<Event> getEventsByUserLocation(String userLocation) {
```

Comment:
Here you should take Principal instead as a parameter and get the current users location in the service